### DISCUSSION TITLE
Make the default dist-tag configurable via package.json

### DISCUSSION THREAD

**Discussion by boutell** [npm/feedback#554] [CLI] [answered]
Many projects must maintain several release series in parallel, as long term support releases, for security patches to earlier series of releases, etc. However the npm publish command will, by default, always tag the most recent publication as `latest`. You can fix this with `--tag` but fatigued developers often mess this up *cough*.

So I propose a `distTag` package.json property. If present, it would be honored as the default dist-tag for `npm publish`, otherwise it would default to `latest` as always. One could always override it with `--tag` just like before.

Consider the simple workflow this enables: if you're maintaining a 2.x series as well as a latest-and-greatest 3.x series, you can pop `"distTag": "v2-lts"` into `package.json`, which you're maintaining separately anyway in such a scenario. Now `npm publish` does the right thing out of the box.

This is something we're implementing in our in-house `np` wrapper. Would be great to see in core.

What do folks think?

**Comment by ljharb** [comment]
I made and use https://npmjs.com/safe-publish-latest in a prepublishOnly script to protect me from this mistake. It’d be great if you filed this as an RFC on the relevant repo so npm can discuss adding it!

**Comment by boutell** [comment, marked as answer]
Oh, that's even better. Scenarios where it didn't suffice would be rare.

I'm not sure where I should file it.


On Tue, Sep 14, 2021 at 11:49 AM Jordan Harband ***@***.***>
wrote:

> I made and use https://npmjs.com/safe-publish-latest in a prepublishOnly
> script to protect me from this mistake. It’d be great if you filed this as
> an RFC on the relevant repo so npm can discuss adding it!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/npm/feedback/discussions/554#discussioncomment-1325118>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27LXM5J7DC64I4QVDF3UB5VIJANCNFSM5EAJPXTA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his

**Reply by ljharb** [reply]
https://github.com/npm/rfcs
